spec/models/schema_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe Schema, type: :model do
+  context 'when created' do
+    subject { described_class.new }
+
+    it 'must have an endpoint and a structure' do
+      expect { subject.save! }.to raise_error(ActiveRecord::RecordInvalid)
+      subject.structure = '{ "field": "value" }'
+      expect { subject.save! }.to raise_error(ActiveRecord::StatementInvalid)
+    end
+
+    it 'must have a non-empty structure' do
+      subject.structure = '{ }'
+      expect { subject.save! }.to raise_error(ActiveRecord::RecordInvalid)
+    end
```

Comment:
Since you have your overall `describe` for the test, and then you wrap everything inside one more context...there is no real reason for that except test output.  So why not change it to be like this:
```ruby
describe class do
  subject { described_class.new }
  describe "Schema errors correctly when" do
    it "does not have an endpoint" do
       expect { subject.save! }.to raise_error(ActiveRecord::RecordInvalid)
    end
    context "#structure" do
      it "catches invalid statements" do
         subject.structure = '{ "field": "value" }'
         expect { subject.save! }.to raise_error(ActiveRecord::StatementInvalid)
      end
      it "cannot have an empty structure" do
        subject.structure = '{ }'
        expect { subject.save! }.to raise_error(ActiveRecord::RecordInvalid)
      end
    end
  end
end
```
Now, that said...I don't like all of the test names I have suggested.  But, this gives you a little bit more future-flexibility of where to place things.  And you also need to still decide where to place the DateTime tests.